### PR TITLE
Feat/pg credential configuration

### PR DESCRIPTION
**Pull request by hieu-openfabric**
[SP-2204](https://open-fabric.atlassian.net/browse/SP-2204)

[SP-2204]: https://open-fabric.atlassian.net/browse/SP-2204?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: templates/admin_options_field_checkbox.php [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<tr valign="top">
+<tr valign="top" data-field-key="<?php echo $field_key; ?>">
```
Comment (tungd) [inline templates/admin_options_field_checkbox.php:1]: This is the main fix we're looking for

File: includes/of_constants.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    /*     TEST_ENV => 'https://auth.dev.openfabric.co', */
+    TEST_ENV => 'https://auth.sandbox.openfabric.co', // sandbox
+    LIVE_ENV => 'https://auth.openfabric.co', // production
+  );
+  
+  const OF_ENDPOINTS = array(
+    /*     TEST_ENV => 'https://api.dev.openfabric.co', */
+    TEST_ENV => 'https://api.sandbox.openfabric.co', // sandbox
+    LIVE_ENV => 'https://api.openfabric.co', // production
+  );
```
Comment (tungd) [inline includes/of_constants.php:16]: Same here, as with the first comment. In general, I'm also very against the style of having a constant file per project. Code should be structured by their domain, not by functions. These constants are going to be used once, when the plugin is first initialized. There's no real reason to have a file like this and have everything depend on it.

**Review by tungd — CHANGES_REQUESTED** [review]
I have the of

File: includes/class-of-public-api.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    return;
+  }
+  
+  require_once(plugin_dir_path(__FILE__) . 'class-of-helpers.php');
+  require_once(plugin_dir_path(__FILE__) . 'of_constants.php');
+  
+  class OF_Public_API {
+    private $endpoint;
+    private $decorators;
+    private $cache_key = 'PG_CONFIGURATION';
```
[the remark this answers is not in the document]
Comment (hieu-openfabric) [reply includes/class-of-public-api.php:12]: Noop, it is a general configuration, it is not related to the payment method id

File: includes/class-of-helpers.php [hunk 4]
```diff
@@ -28,4 +30,106 @@ static function log($data, $key = 'debug') {
...
+      if ($httpStatusCode < 200 || $httpStatusCode > 299) {
+          $errorMessage = $body->details[0]->message ?? $body->Message;
+          throw new Exception($errorMessage);
+      }
+
+      $response_body = wp_remote_retrieve_body($response);
+      return json_decode($response_body, true);
+  }
+  
+  static function build_payment_gateway_settings(
```
[the remark this answers is not in the document]
Comment (hieu-openfabric) [reply includes/class-of-helpers.php:47]: This is util function only, I think we don't need to break it down to other class

**Review by tungd — APPROVED** [review]

**Pull request merged**